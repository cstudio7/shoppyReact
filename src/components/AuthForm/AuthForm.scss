.formContainer {
  margin: auto;
  margin-top: -30%;
  padding: 5%;
}
.ui.button.loginButton {
  color: #fff;
  background: #ff6300;
  width: 100%;
}
.ui.form.loginForm {
  border-top: 1px solid #f0f0f0;
  padding-top: 12%;
}
.loginField {
  padding-bottom: 5%;
}
.customer {
  display: none;
}
@media only screen and (min-width: 600px) {
  .loginFormColumns {
    margin: 0 auto;
    margin-left: 30%;
  }
  .loginField {
    padding-bottom: 2%;
  }
  .formContainer {
    padding-top: 0;
    padding-bottom: 0;
    margin-top: -10%;
  }
  .customer {
    display: inline-block;
    max-height: 100vh;
    margin-left: 5%;
    margin-top: -5%;
  }
  .storeLogo {
    max-width: 50%;
    padding-left: 5%;
  }
}
@media only screen and (min-width: 800px) {
  .customer {
    margin-top: -15%;
  }
}
