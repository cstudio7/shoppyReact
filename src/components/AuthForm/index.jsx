import React, { Component, Fragment } from 'react';
import { Button, Form } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import logo from './images/logo.png';
import background from './images/background.png';
import { userLogin } from '../../actions/authAction';
import { getUserInfo } from '../../helpers/jwtHelper';
import '../../partial-styles/responsive.scss';
import './AuthForm.scss';

class AuthForm extends Component {
  state = {
    loginDetails: {
      email: '',
      password: ''
    }
  };

  handleLoginSubmit = (event) => {
    event.preventDefault();
    this.props.userLogin(this.state.loginDetails);
  };

  handleChange = (event) => {
    const { loginDetails } = this.state;
    const { type, value } = event.target;
    loginDetails[type] = value;
    this.setState({
      loginDetails
    });
  };

  createLoginForm = () => (
    <Form className="loginForm" onSubmit={this.handleLoginSubmit}>
      <Form.Field className="loginField">
        <input
          placeholder="Email"
          id="userEmail"
          type="email"
          required
          onChange={this.handleChange}
        />
      </Form.Field>
      <Form.Field className="loginField">
        <input
          placeholder="Password"
          id="userPassword"
          type="password"
          required
          onChange={this.handleChange}
        />
      </Form.Field>
      <Button type="submit" className="loginButton">
        Submit
      </Button>
    </Form>
  );

  render() {
    const userInfo = getUserInfo();
    if (userInfo) {
      return <Redirect to="/" />;
    }
    return (
      <Fragment>
        <img src={logo} alt="store logo" className="storeLogo" />
        <div className="row formContainer">
          <div className="col-5 col-s-5 ">{this.createLoginForm()}</div>
          <img src={background} alt="Background graphics" className="col-5 col-s-5 customer" />
        </div>
      </Fragment>
    );
  }
}

AuthForm.propTypes = {
  userLogin: PropTypes.func.isRequired,
  userLoginStatus: PropTypes.shape({})
};

AuthForm.defaultProps = {
  userLoginStatus: {}
};

const mapStateToProps = state => ({
  userLoginStatus: state.userLogin
});
export { AuthForm as AuthFormPage };
export default connect(
  mapStateToProps,
  { userLogin }
)(AuthForm);
