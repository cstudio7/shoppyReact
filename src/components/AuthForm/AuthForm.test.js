import React from 'react';
import { shallow } from 'enzyme';
import AuthForm from './index';

describe('Test the Login Form Component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<AuthForm />);
    expect(wrapper).toMatchSnapshot();
  });
});
