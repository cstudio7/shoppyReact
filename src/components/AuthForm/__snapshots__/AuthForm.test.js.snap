// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Login Form Component should match snapshot 1`] = `
<Fragment>
  <img
    alt="store logo"
    className="storeLogo"
    src="test-file-stub"
  />
  <div
    className="row formContainer"
  >
    <div
      className="col-5 col-s-5 "
    >
      <Form
        as="form"
        className="loginForm"
        onSubmit={[Function]}
      >
        <FormField
          className="loginField"
        >
          <input
            id="userEmail"
            onChange={[Function]}
            placeholder="Email"
            required={true}
            type="email"
          />
        </FormField>
        <FormField
          className="loginField"
        >
          <input
            id="userPassword"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
          />
        </FormField>
        <Button
          as="button"
          className="loginButton"
          type="submit"
        >
          Submit
        </Button>
      </Form>
    </div>
    <img
      alt="Background graphics"
      className="col-5 col-s-5 customer"
      src="test-file-stub"
    />
  </div>
</Fragment>
`;

exports[`Test the Login Form Component should match snapshot 2`] = `
<Redirect
  push={false}
  to="/"
/>
`;
